src/lib/shared/api/not-null.ts

```diff
@@ -0,0 +1,3 @@
+export function notNull<ArgumentType>(argument: ArgumentType | null) {
```

Comment:
It would have helped with type inference and flow analysis if the function was declared to be a type guard by setting the appropriate return type. Check this example inspired by [the docs](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#assertion-functions):
```ts
function isDefined<T>(val: T): val is NonNullable<T> {
  return !(val === undefined || val === null);
}
```